File: 20A40/cir-villalobos/completos/e40/Employee.java
public class Employee {

    private String name;
    private String adress;
    private double salary;

    public boolean setName(String nam){
        if(!(nam.length() > 0))
            return false;
        this.name = nam;
        return true;
    }

    public boolean setAdress(String add){
        if(!(add.length() > 0))
            return false;
        this.adress = add;
        return true;
    }

    public boolean setSalary(double sal){
        if(!(sal >= 0))
            return false;
        this.salary = sal;
        return true;
    }

    public String getName()   {    return this.name;      }
    public String getAdress() {    return this.adress;    }
    public double getSalary() {    return this.salary;    }

    public String toString(){
        String concat = "Name: " + name + "\tSalary: " +
                        salary + "\tAdress: " + adress;
        return concat;
    }
}

File: 20A40/cir-agus/EJER_40/Employee.py
class Employee:
    def __init__(self, name, adress, salary) -> None:
        self.__name = name
        self.__adrs = adress
        self.__sala = salary
        
    def setName(self, nam) -> bool:
        if len(nam) > 0:
            self.__name = nam
            return True
        else: return False
    
    def setAdress(self, adr) -> bool:
        if len(adr) > 0:
            self.__adrs = adr
            return True
        else: return False
    
    def setSalary(self, saly) -> bool:
        if saly > 0:
            self.__sala = saly
            return True
        else: return False
    
    def getName(self) -> str:   return self.__name
    def getAdress(self) -> str:   return self.__adrs
    def getSalary(self) -> float: return self.__sala
    
    def toString(self) -> str:
        cadena = f"Nombre: {self.__name}\n"
        cadena += f"Direccion: {self.__adrs}\n"
        cadena += f"Salario: {self.__sala}"
        return cadena

"""Falta estudio
    def __init_subclass__(cls,name, adress, salary) -> None:
        cls.__name = name
        cls.__adrs = adress
        cls.__sala = salary
        return cls.__init__(Employee,name,adress,salary)
    """